### PR TITLE
Include rendering time in Perf Vis tests

### PR DESCRIPTION
**Pull request by mmisol**
### Purpose

Include rendering time in performance visualization tests.

### Declarations

Check these if you believe they are true

- [x] The codebase is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [x] User facing strings, if any, are extracted into `*.resx` files
- [x] All tests pass using the self-service CI.
- [x] Snapshot of UI changes, if any.
- [x] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [x] This PR modifies some build requirements and the readme is updated

### Reviewers

@aparajit-pratap 

### FYIs

@DynamoDS/dynamo

### CODE REVIEW

File: src/DynamoCoreWpf/ViewModels/Watch3D/HelixWatch3DViewModel.cs [hunk 1]
```diff
@@ -1030,15 +1034,6 @@ protected override bool CanToggleCanNavigateBackground(object parameter)
 
         internal void ComputeFrameUpdate()
         {
-#if DEBUG
-            if (renderTimer.IsRunning)
```
Comment (aparajit-pratap) [inline src/DynamoCoreWpf/ViewModels/Watch3D/HelixWatch3DViewModel.cs:1034]: Why was this deleted? I saw that control still went here some times during my debugging.

File: src/DynamoCoreWpf/ViewModels/Watch3D/HelixWatch3DViewModel.cs [hunk 2]
```diff
@@ -855,10 +856,13 @@ public override void GenerateViewGeometryFromRenderPackagesAndRequestUpdate(Rend
...
-            Debug.WriteLine(string.Format("RENDER: {0} ellapsed for compiling assets for rendering.", renderTimer.Elapsed));
-            renderTimer.Reset();
-            renderTimer.Start();
+            // Defer stopping the timer until after the rendering has occurred
+            Dispatcher.CurrentDispatcher.BeginInvoke(DispatcherPriority.Loaded, new Action(() =>
+            {
+                renderTimer.Stop();
+                Debug.WriteLine(string.Format("RENDER: {0} ellapsed time rendering.", renderTimer.Elapsed));
+                renderTimer.Restart();
+            }));
```
Comment (aparajit-pratap) [inline src/DynamoCoreWpf/ViewModels/Watch3D/HelixWatch3DViewModel.cs:865]: 👍

File: src/VisualizationTests/PerformanceVisualizationTests.cs [hunk 3]
```diff
@@ -18,14 +18,16 @@ namespace WpfVisualizationTests
...
+        const int SamplingAmount = 3;
 
         [Test]
         [TestCaseSource("PerformanceTestSource")]
         public void TestVisualizationPerformance(string filePath)
         {
             ViewModel.OpenCommand.Execute(filePath);
+            DispatcherUtil.DoEvents();
             RunCurrentModel();
+            DispatcherUtil.DoEvents();
```
Comment (aparajit-pratap) [inline src/VisualizationTests/PerformanceVisualizationTests.cs:30]: Do these tests now run as part of the other performance benchmark tests? It's been a long time since I ran those tests locally. It would be interesting to see what are the render time results from running these.

File: src/DynamoCoreWpf/ViewModels/Watch3D/HelixWatch3DViewModel.cs (hunk 1 above)
Comment (mmisol) [reply src/DynamoCoreWpf/ViewModels/Watch3D/HelixWatch3DViewModel.cs:1034]: Since we are scheduling the timer stop for after rendering occurs, this was actually disrupting the elapsed time obtained, because it used the same timer.
While we could change this to use another timer instead of deleting it, I didn't see this provide accurate results during my tests, so I just deleted it. Also, the new measure includes what this is supposed to include, so I don't think we want to track this separately.

File: src/VisualizationTests/PerformanceVisualizationTests.cs (hunk 3 above)
Comment (mmisol) [reply src/VisualizationTests/PerformanceVisualizationTests.cs:30]: These tests relation with performance benchmark tests is only that by default they use the same graphs. These are aimed to just measure rendering time, which is something that gets mixed with a lot of other stuff in the performance benchmark.
Also these tests are not really a tool, you need to do some tweaking and do some extra steps to get results out of them.

Comment (aparajit-pratap) [reply src/VisualizationTests/PerformanceVisualizationTests.cs:30]: Can you include comments on how to run these tests and see the timing results?

Comment (mmisol) [reply src/VisualizationTests/PerformanceVisualizationTests.cs:30]: It's in the top of this file, in the class comment.

Comment (aparajit-pratap) [reply src/VisualizationTests/PerformanceVisualizationTests.cs:30]: `For that, you should make sure the renderTimer Stopwatch in HelixWatch3DViewModel is enabled and logging to Console. Output should be redirected to a file.`

Does one need to add a `TextWriterTraceListener` for this? Can you give an example of how would someone carry out the steps above?

Comment (mmisol) [reply src/VisualizationTests/PerformanceVisualizationTests.cs:30]: I haven't explored using `TextWriterTraceListener`. For my testing I just run the test using nunit from the command line and redirect output to a file using `>`.
If you want to run in `Release` mode, you need to do changes to make sure the Stopwatch is enabled. Removing the `#if DEBUG` surrounding the timer where it's defined and used should be needed. Also, to guarantee you are logging to Console in Release, you should use `Console.WriteLine` rather than `Debug.WriteLine`.

Comment (aparajit-pratap) [reply src/VisualizationTests/PerformanceVisualizationTests.cs:30]: Thanks. Should we just change `Debug.WriteLine` to `Console.WriteLine` in that case?

Comment (mmisol) [reply src/VisualizationTests/PerformanceVisualizationTests.cs:30]: I think `Debug.WriteLine` should be considered a better practice in general, even if under a `#if DEBUG` flag. The possibility to add trace listeners is there, even if we don't use it today. In general we are using `Debug.WriteLine` in other places to log stuff, so I think it's more consistent if we keep it that way.
